### PR TITLE
Use WaitAsync

### PR DESCRIPTION
**Pull request by WeihanLi**
Contributes to #1055 

Use WaitAsync task timeout extensions instead of TimeoutAfter for .NET 6

### CODE REVIEW

File: test/ReverseProxy.Tests/Common/TaskExtensions.cs [hunk 1]
```diff
@@ -21,24 +21,41 @@ internal static class TaskExtensions
...
+            return task.AsTask().WaitAsync(DefaultTimeoutTimeSpan);
+#else
             return task.AsTask().TimeoutAfter(DefaultTimeoutTimeSpan);
+#endif
         }
 
         public static Task<T> DefaultTimeout<T>(this Task<T> task)
         {
+#if NET6_0_OR_GREATER
+            return task.WaitAsync(DefaultTimeoutTimeSpan);
```
Comment (Tratcher) [inline test/ReverseProxy.Tests/Common/TaskExtensions.cs:43]: What does this throw when it expires?  Before it was throwing a specific exception with details:
`throw new TimeoutException(CreateMessage(timeout, filePath, lineNumber));`

Comment (MihaZupan) [reply test/ReverseProxy.Tests/Common/TaskExtensions.cs:43]: Might be better to copy what was done in AspNetCore:
https://github.com/dotnet/aspnetcore/blob/dfcf1adee2c0f6c89ecf66a8791e550bce124698/src/Shared/TaskExtensions.cs#L77-L108

Comment (WeihanLi) [reply test/ReverseProxy.Tests/Common/TaskExtensions.cs:43]: It would throw `TimeoutException` also for a timeout

Comment (WeihanLi) [reply test/ReverseProxy.Tests/Common/TaskExtensions.cs]: Got it, thanks @Tratcher

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**